File: plugins/OrchidPosts/src/main/java/com/eden/orchid/posts/permalink/pathTypes/CategoryPathType.java
package com.eden.orchid.posts.permalink.pathTypes;

import com.eden.orchid.posts.pages.PostPage;
import com.eden.orchid.posts.permalink.PermalinkPathType;

import javax.inject.Inject;

public class CategoryPathType extends PermalinkPathType {

    @Inject
    public CategoryPathType() {
        this.priority = 100;
    }

    @Override
    public boolean acceptsKey(PostPage post, String key) {
        return key.equals("category");
    }

    @Override
    public String format(PostPage post, String key) {
        return post.getCategory();
    }

}

File: plugins/OrchidPosts/src/main/kotlin/com/eden/orchid/posts/permalink/pathTypes/CategoryPathType.kt
package com.eden.orchid.posts.permalink.pathTypes

import com.eden.orchid.api.theme.pages.OrchidPage
import com.eden.orchid.api.theme.permalinks.PermalinkPathType
import com.eden.orchid.posts.pages.PostPage
import javax.inject.Inject

class CategoryPathType @Inject
constructor() : PermalinkPathType(100) {

    override fun acceptsKey(page: OrchidPage, key: String): Boolean {
        return key == "category" && page is PostPage
    }

    override fun format(page: OrchidPage, key: String): String? {
        if (page is PostPage) {
            return if(page.categoryModel.key != null) page.categoryModel.path else ""
        }

        return null
    }

}

